colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -296,4 +300,13 @@ extends Controller[O,I](codec, ControllerConfig(config.pendingBufferSize, config
     }
   }
 
+
+  override def idleCheck(period: Duration) {
+    super.idleCheck(period)
+
+    if (sentBuffer.size > 0 && sentBuffer.front.isTimedOut(System.currentTimeMillis)) {
+      // the oldest sent message has expired with no response - shutdown the connection
+      disconnect()
```

Comment:
I think we need to think a bit more about if this is the right thing to do.  In particular, do we want the client to attempt to reconnect?  I think we do want to reconnect, so then we don't want to just call `disconnect()` because that treats it like the user is consciously deciding to close the connection and no reconnect is attempted.

Instead we might want to handle this more like an error, in which case we should have the client basically go through the same logic as if `connectionLost` is called.

Also at the very least we should be more vocal about this happening.  Just calling disconnect will close the connection silently without any indication as to why.